src/main/java/uk/gov/hmcts/sptribs/caseworker/helper/RecordListHelper.java

```diff
@@ -126,25 +130,28 @@ public Listing checkAndUpdateVenueInformation(Listing listing) {
     }
 
     public Listing checkAndUpdateVenueInformationSummary(Listing listing) {
-        if ((null == listing.getVenueNotListedOption()
+        if ((listing.getVenueNotListedOption() == null
             || !listing.getVenueNotListedOption().contains(VenueNotListed.VENUE_NOT_LISTED))
-            && null != listing.getReadOnlyHearingVenueName() && !listing.getReadOnlyHearingVenueName().isEmpty()) {
+            && listing.getReadOnlyHearingVenueName() != null && !listing.getReadOnlyHearingVenueName().isEmpty()) {
             listing.setHearingVenueNameAndAddress(listing.getReadOnlyHearingVenueName());
         }
         return listing;
     }
 
     public Listing saveSummary(CaseData caseData) {
-        caseData.getListing().setHearingFormat(caseData.getListing().getHearingFormat());
-        caseData.getListing().setHearingType(caseData.getListing().getHearingType());
-        caseData.getListing().getSummary().setSubjectName(caseData.getCicCase().getFullName());
-        caseData.setCurrentEvent("");
-        if (null != caseData.getListing()
-            && null != caseData.getListing().getNumberOfDays()
-            && caseData.getListing().getNumberOfDays().equals(YesOrNo.NO)) {
-            caseData.getListing().setAdditionalHearingDate(null);
+        final Listing listing = caseData.getListing();
```

Comment:
Refactored slightly for better readability and testing. Though further improvement could be to refactor all `checkAndUpdateVenueInformation`, `checkAndUpdateVenueInformationSummary` and `SaveSummary` to all be void methods as they only  modify the `caseData.getListing()` object.